chapters/11-version-control-workflow.Rmd

```diff
@@ -1,18 +1,244 @@
-# Version control: Workflows around using GitHub (part 2) {#github-workflows}
+# Workflows when using GitHub {#github-workflows}
 
-- Make use of R builtin tools like the usethis pull request helpers
-  (and learning how to make use of this yourself)
-- E.g. with pr_* functions from usethis
-- Using this workflow as a team or as a contributor
+<!-- Brief summary of list of actions/code/concepts done -->
+
+- Exercises: Conceptual, how can people effectively collaborate on a project?
+    - Maybe get them to create a vignette article with their answers/thoughts? 
+    Articles that are ignored from package build. (we discussed this a bit
+    already, needs more discussion)
+- Branching/forks
+    - Text: what is it, power of it, features.
+- Pull Requests
+    - Exercises: Reading, comprehension, and conceptual understanding (non-coding)
+- Security:
+    - Text: why its important, how to do it.
+    - Code:
+        - usethis::git_sitrep()
+        - usethis::gh_token_help()
+        - usethis::create_github_token()
+    - Text: Save PAT in password manager (they should have this anyway, but this forces them)
+    - Code: 
+        - gitcreds::gitcreds_set(), everytime computer restarts, only saved for session
+    - Exercises: Conceptual understanding of PAT, tokens, and password managers (non-coding)
+    - Exercises: Setting up PAT, saving PAT, using gitcreds at start of session
+- usethis pr_* helpers
+    - Text: Why use these over GitHub interface
+- As contributor/sole developer/as team
+    - usethis::pr_init()
+        - Make changes to Contributing about using pr helpers
+    - usethis::pr_push()
+    - usethis::pr_view()
+    - usethis::pr_pause()
+    - Exercise: Make an edit on the PR changed files using GitHubs edit interface.
+    - Exercise: Create another PR for adding ...FIXME.
+    - usethis::pr_resume()
+    - usethis::pr_pull()
+    - usethis::pr_merge_main()
+    - usethis::pr_push()
+    - PR has been merged
+    - gert::git_branch_list() # Show list of created branches
+    - usethis::pr_finish()
+    - Exercise: Do the same process for the other PR you had created.
+    - gert::git_branch_list() # Show list of created branches
+    - usethis::pr_init()
+    - usethis::pr_forget()
+    - gert::git_branch_list() # Show list of created branches
+- As maintainer, reviewer, team lead. (Hard to do exercises for this... need to
```

Comment:
Could we have a GitHub Action that simulates this behaviour for solo learners? So that they can practice it. If it's overkill I understand 😄

Reply:
Hmm, interesting idea. Let's make an issue and think about it.